pennylane/devices/null_qubit.py

```diff
@@ -137,52 +139,47 @@ class NullQubit(Device):
             (``['aux_wire', 'q1', 'q2']``). Default ``None`` if not specified.
         shots (int, Sequence[int], Sequence[Union[int, Sequence[int]]]): The default number of shots
             to use in executions involving this device.
+        operations (Optional[Iterable[str, type]]): a target gateset for the device.
+        target_device (Optional[pennylane.devices.Device]): a device to mimic the preprocessing of. Must obey the new device interface.
+        assume_no_broadcasting=False (bool): If ``True``, we always assume no parameter batching exists. Useful
+            for profiling and benchmarking.
 
     **Example:**
 
-    .. code-block:: python
-
-        qs = qml.tape.QuantumScript(
-            [qml.Hadamard(0), qml.CNOT([0, 1])],
-            [qml.expval(qml.PauliZ(0)), qml.probs()],
-        )
-        qscripts = [qs, qs, qs]
-
-    >>> dev = NullQubit()
-    >>> program, execution_config = dev.preprocess()
-    >>> new_batch, post_processing_fn = program(qscripts)
-    >>> results = dev.execute(new_batch, execution_config=execution_config)
-    >>> post_processing_fn(results)
-    ((array(0.), array([1., 0., 0., 0.])),
-     (array(0.), array([1., 0., 0., 0.])),
-     (array(0.), array([1., 0., 0., 0.])))
-
-
-    This device currently supports (trivial) derivatives:
-
-    >>> from pennylane.devices import ExecutionConfig
-    >>> dev.supports_derivatives(ExecutionConfig(gradient_method="device"))
-    True
-
     This device can be used to track resource usage:
 
     .. code-block:: python
 
-        n_layers = 50
-        n_wires = 100
-        shape = qml.StronglyEntanglingLayers.shape(n_layers=n_layers, n_wires=n_wires)
+        def track_entangling_layers(device):
+            n_layers = 50
+            n_wires = 100
+            shape = qml.StronglyEntanglingLayers.shape(n_layers=n_layers, n_wires=n_wires)
 
-        @qml.qnode(dev)
-        def circuit(params):
-            qml.StronglyEntanglingLayers(params, wires=range(n_wires))
-            return [qml.expval(qml.Z(i)) for i in range(n_wires)]
+            @qml.qnode(device)
+            def circuit(params):
+                qml.StronglyEntanglingLayers(params, wires=range(n_wires))
+                return [qml.expval(qml.Z(i)) for i in range(n_wires)]
 
-        params = np.random.random(shape)
+            params = np.random.random(shape)
 
-        with qml.Tracker(dev) as tracker:
-            circuit(params)
+            with qml.Tracker(device) as tracker:
+                circuit(params)
+            return tracker
 
-    >>> tracker.history["resources"][0]
+    >>> track_entangling_layers(qml.device('null.qubit')).latest['resources']
```

Comment:
```suggestion
    >>> dev = qml.device('null.qubit')
    >>> track_entangling_layers(dev).latest['resources']
```
Looks more readable and less crowded this way.